addons/payment_stripe/static/src/js/stripe_mixin.js

```diff
@@ -2,85 +2,69 @@
+     * @override method from payment.payment_form
      * @private
-     * @param {string} code - The code of the selected payment option's provider.
-     * @param {number} paymentOptionId - The id of the selected payment option.
-     * @param {string} flow - The online payment flow of the selected payment option.
+     * @param {number} providerId - The id of the selected payment option's provider.
+     * @param {string} providerCode - The code of the selected payment option's provider.
+     * @param {number} paymentOptionId - The id of the selected payment option
+     * @param {string} paymentMethodCode - The code of the selected payment method, if any.
+     * @param {string} flow - The online payment flow of the selected payment option
      * @return {void}
      */
-    _prepareInlineForm(code, paymentOptionId, flow) {
-        if (code !== 'stripe') {
+    async _prepareInlineForm(providerId, providerCode, paymentOptionId, paymentMethodCode, flow) {
+        if (providerCode !== 'stripe') {
             this._super(...arguments);
             return;
         }
 
+        // Check if instantiation of the element is needed.
+        this.stripePaymentElements ??= {}; // Store the element of each instantiated payment method.
         // Check if instantiation of the element is needed.
         if (flow === 'token') {
             return; // No elements for tokens.
-        } else if (this.stripeElements && this.stripeElements.providerId === paymentOptionId) {
+        } else if (this.stripePaymentElements[paymentOptionId]) {
             this._setPaymentFlow('direct'); // Overwrite the flow even if no re-instantiation.
             return; // Don't re-instantiate if already done for this provider.
         }
 
         // Overwrite the flow of the select payment option.
         this._setPaymentFlow('direct');
 
-        // Extract and deserialize the inline form values.
-        const stripeInlineForm = document.getElementById(
-            `o_stripe_${this.formType}_element_container_${paymentOptionId}`
-        );
-        this.stripeInlineFormValues = JSON.parse(stripeInlineForm.dataset['inlineFormValues']);
+        const radio = document.querySelector('input[name="o_payment_radio"]:checked');
+        const inlineForm = this._getInlineForm(radio);
+        const stripeInlineForm = inlineForm.querySelector('[name="o_stripe_element_container"]');
+        this.isValidation = Boolean(this.txContext['mode'] === 'validation');
+        this.stripeInlineFormValues = JSON.parse(radio.dataset['inlineFormValues']);
+        this.stripeInlineFormValues['paymentMethodTypes'] = [paymentMethodCode]
 
         // Instantiate the payment element.
-        this.stripeJS = Stripe(
-            this.stripeInlineFormValues['publishable_key'],
-            // The values required by Stripe Connect are inserted into the dataset.
-            new StripeOptions()._prepareStripeOptions(stripeInlineForm.dataset),
```

Comment:
We need those informations for Stripe Connect, this code is overridden in our internal repo :)